### DISCUSSION TITLE
Very different calculated consumption

### DISCUSSION THREAD

**Discussion by vzhivkov** [CarlosCuezva/dashboards-Grafana-Teslamate#23] [General]
Hello,
I still have active TeslaFi account and I already put into action Teslamate. I see very different consumption for a given drive. For example today I have 2 drives:
1st in TeslaFi is 109 Wh/km for 4.23 km , and in Drives v2 is 154 Wh/km for 4.2 km
2nd in TeslaFi si 113 Wh/km for 5.05 km and in Drives v2 is 129 Wh/km for 5.0 km
yesterday I had a long one including highway and non-highway roads
that one in TeslaFi was 157 Wh/km for 90.26 km Driven and in Drives v2 that same drive was 177 Wh/km for 90.3 km.

I guess Drives v2 uses the same formula as original Drives in Teslamate where I have same wrong consumption. Could that formula be changed somehow to get results closer to the one shown on Tesla screen?

**Comment by CarlosCuezva** [comment]
Hi!

Correct, I use the same calculation in Drives v2. The problem with the formula is that neither the consumption of the trip nor the efficiency of the car is stored in the database, so it can only be calculated from the data that we do have.
